stable_baselines3/common/utils.py

```diff
@@ -252,7 +252,7 @@ def is_vectorized_discrete_observation(observation: np.ndarray, observation_spac
     :param observation_space: the observation space
     :return: whether the given observation is vectorized or not
     """
-    if observation.shape == ():  # A numpy array of a number, has shape empty tuple '()'
+    if isinstance(observation, int) or observation.shape == ():  # A numpy array of a number, has shape empty tuple '()'
```

Comment:
@Miffyli this is the issue from https://github.com/DLR-RM/stable-baselines3/issues/586
we were not expecting int but it is possible as the observation here is before pre-processing.
Luckily, the fix is simple ;)